query/compile.go

```diff
@@ -550,6 +556,161 @@ func (c *compiledField) compileMovingAverage(args []influxql.Expr) error {
 	}
 }
 
+func (c *compiledField) compileExponentialMovingAverage(name string, args []influxql.Expr) error {
+	if got := len(args) - 1; got < 1 || got > 3 {
+		return fmt.Errorf("invalid number of arguments for %s, expected at least 1 but no more than 3, got %d", name, got)
+	}
+
+	switch arg1 := args[1].(type) {
+	case *influxql.IntegerLiteral:
+		if arg1.Val < 1 {
+			return fmt.Errorf("%s period must be greater than or equal to 1", name)
+		}
+	default:
+		return fmt.Errorf("%s period must be an integer", name)
+	}
+
+	if len(args) >= 3 {
+		switch arg2 := args[2].(type) {
+		case *influxql.IntegerLiteral:
+			if name == "triple_exponential_average" && arg2.Val < 1 && arg2.Val != -1 {
+				return fmt.Errorf("%s hold period must be greater than or equal to 1", name)
+			}
+			if arg2.Val < 0 && arg2.Val != -1 {
+				return fmt.Errorf("%s hold period must be greater than or equal to 0", name)
+			}
+		default:
+			return fmt.Errorf("%s hold period must be an integer", name)
+		}
+	}
+
+	if len(args) >= 4 {
+		switch arg3 := args[3].(type) {
+		case *influxql.StringLiteral:
+			switch arg3.Val {
+			case "exponential_moving_average":
```

Comment:
For this, you can put the two accepted strings on the same line.

```
case "exponential_moving_average", "average":
```

Can we also name these "simple" and "exponential"? I don't like the length of exponential moving average and I would like it if we used the prefix for the warmup type. I notice that `EMA` stands for "exponential moving average" and `SMA` stands for "simple moving average" so labeling the field as "this is the moving average type" would mean that we don't need to use the longer name and I think it would reduce confusion.

I understand your concern so please push back on this if you feel that the longer name is very important.